Tests/js/views/services/assets/ez-languageselectionboxviewservice-tests.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-languageselectionboxviewservice-tests', function (Y) {
+    var getViewParametersTest,
+        Mock = Y.Mock, Assert = Y.Assert;
+
+    getViewParametersTest = new Y.Test.Case({
+        name: "eZ Language Selection Box View Service getViewParameters test",
+
+        setUp: function () {
+            this.app = new Mock();
+            this.systemLanguageList = {
+                'eng-EN': {languageCode: 'eng-EN', name: 'EN'},
+                'pol-PL': {languageCode: 'pol-PL', name: 'PL'}
+            };
+
+            Mock.expect(this.app, {
+                method: 'get',
+                args: ['systemLanguageList'],
+                returns: this.systemLanguageList
+            });
+
+            this.service = new Y.eZ.LanguageSelectionBoxViewService({
+                app: this. app
```

Comment:
while at it, you should also check that `getViewParameters` keeps the values in the `parameters` attribute as well.